alacritty_terminal/src/config/mod.rs

```diff
@@ -209,8 +209,9 @@ impl<T> Config<T> {
     }
 
     #[inline]
-    pub fn working_directory(&self) -> &Option<PathBuf> {
-        &self.working_directory
+    pub fn working_directory(&self) -> Option<&Path> {
+        // TODO: On Rust 1.40 use self.working_directory.deref()
+        self.working_directory.as_ref().map(|path_buf| path_buf.as_ref())
```

Comment:
`Option<&Path>` is one pointer indirection less than `&Option<PathBuf>`, so I think it's better style to return it in this case.

Regarding the TODO, I guess we are some way off setting a minimum Rust version of 1.40 so we can use `Option::deref()`? I also considered using the `rustversion` crate instead of a TODO.

Reply:
Why not just use `self.warking_directory.map(|path_buf| path_buf.as_path())`?

I think that's more descriptive than both the existing and the proposed 1.40 version and it should work just as well?

Reply:
Can do. Though it would have to be 
 
   self.working_directory.as_ref().map(|path_buf| path_buf.as_path())

Because `Option::map` consumes `self`.

Reply:
Sounds good to me.